### PR TITLE
fix: add try-catch block to authentication manager

### PR DESCRIPTION
**Pull request by swetabar**
## Related Issues

Add try-catch block to authentication-manager to try and figure out the issue with ScopedApiKeyHandler

### CODE REVIEW

File: packages/spacecat-shared-http-utils/src/auth/authentication-manager.js [hunk 1]
```diff
@@ -45,8 +45,13 @@ export default class AuthenticationManager {
...
       this.log.debug(`Trying to authenticate with ${handler.name}`);
 
-      // eslint-disable-next-line no-await-in-loop
-      const authInfo = await handler.checkAuth(request, context);
+      let authInfo;
+      try {
+        // eslint-disable-next-line no-await-in-loop
+        authInfo = await handler.checkAuth(request, context);
+      } catch (error) {
+        this.log.error(`Failed to authenticate with ${handler.name}: ${error.message}`);
```
Comment (blefebvre) [inline packages/spacecat-shared-http-utils/src/auth/authentication-manager.js:53]: To log the stack trace as well:
```suggestion
        this.log.error(`Failed to authenticate with ${handler.name}:`, error);
```

File: packages/spacecat-shared-http-utils/src/auth/handlers/scoped-api-key.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ export default class ScopedApiKeyHandler extends AbstractHandler {
       this.log(`No API key entity found in the data layer for the provided API key: ${apiKeyFromHeader}`, 'error');
       return null;
     }
+    this.log(`API Key entity: ${apiKeyEntity}`, 'debug');
```
Comment (blefebvre) [inline packages/spacecat-shared-http-utils/src/auth/handlers/scoped-api-key.js:49]: Should this object be stringified?

**Review by blefebvre — APPROVED** [review]
Minor item with the log statement.

Is an additional test case needed to hit the `catch` block?

File: packages/spacecat-shared-http-utils/src/auth/handlers/scoped-api-key.js (hunk 2 above)
Comment (swetabar) [reply packages/spacecat-shared-http-utils/src/auth/handlers/scoped-api-key.js]: Removing these logs as we might not need them anymore.

**Comment by swetabar** [comment]
> Minor item with the log statement.
> 
> Is an additional test case needed to hit the `catch` block?

Yes. Added the additional test case to fix the test errors.

**Pull request merged**